[ADD] inventory: valuation by lots (#10858)
odoo/documentation: content/applications/inventory_and_mrp/inventory/product_management/inventory_valuation/valuation_by_lots.rst

================================
Valuation by lots/serial numbers
================================

Track :doc:`inventory valuation <using_inventory_valuation>` by :doc:`lots or serial numbers
<../../product_management/product_tracking>` to:

> #. Compare and differentiate purchasing cost based on lot/SN.

Author: oh wow I definitely forgot to add an example for these! Thank you for the reminder (:

#. Track the actual cost of manufactured products based on the real cost of each tracked component
   used.

> #. Depreciate specific lots/SNs when they :doc:`sit in stock for too long

Author: ```suggestion
#. Depreciate specific lot or serial numbers when they :doc:`sit in stock for too long
```

   <../../warehouses_storage/reporting/aging>`.

Configuration
=============

To enable valuation by lots or serial numbers, begin by enabling the :ref:`Lots and Serial Numbers
feature <inventory/product_management/enable-lot-serial>`. After that, go to
:menuselection:`Inventory app --> Products --> Products` and select the desired product.

On the product form, in the :guilabel:`Category` field, choose a product category. Then, choose the

> :icon:`oi-arrow-right` :guilabel:`(right arrow)` icon to open the product category's settings page.

Author: ```suggestion
:icon:`oi-arrow-right` :guilabel:`(Internal Link)` icon to open the product category's settings page.
```

In the :guilabel:`Inventory Valuation` section, choose the desired :guilabel:`Costing Method`.

> .. seealso::

Author: This is a REALLY good point. Thank you (:

   :ref:`Costing methods <inventory/warehouses_storage/valuation-on-product-category>`

After configuring the costing method, activate the product to be tracked by lots or serial numbers
by ticking the :guilabel:`Track Inventory` checkbox. Then, click the field that becomes visible and
choose either :guilabel:`By Lots` or :guilabel:`By Unique Serial Number` from the resulting
drop-down menu.

Doing so makes the :guilabel:`Valuation by Lot/Serial number` checkbox appear below it. Tick it, and
the configuration to track valuation by lot or serial numbers is complete.

> .. figure:: valuation_by_lots/product-form.png

Author: Thank you :D

items, jump to their dedicated sections:

#. :ref:`Purchase and selling products <inventory/product_management/valuation-cost-example>`: cost
   is updated based on the *product category's* costing method. See the :ref:`Costing Methods
   section <inventory/warehouses_storage/costing_methods>`
#. Inventory adjustment to :ref:`create new lot/serial number
   <inventory/product_management/valuation-cost-new>`: value is assigned to the cost from the
   product form.
#. Inventory adjustment to update quantities for an :ref:`existing lot/serial number
   <inventory/product_management/valuation-cost-existing>`: value is assigned to the cost of the
   most recent value for that lot/serial number.

.. _inventory/product_management/valuation-cost-example:

Calculate cost on product form
------------------------------

A product, tracked by lots using the average cost valuation method (AVCO), is received:

.. list-table::
   :header-rows: 1
   :stub-columns: 1

   * -
     - Quantity
     - Lot number
     - Average cost on product form
   * - Receive at $10
     - 1.00
     - LOT 1
     - $10
   * - Receive at $20
     - 1.00
     - LOT 2
     - $15

> .. figure:: valuation_by_lots/lip-gloss.png

Author: Thank you for these suggestions! Implemented them in my revision (:

lot/serial number form page.

Valuation report
----------------

Display the valuation of lots and serial numbers in the database by going to
:menuselection:`Inventory app --> Reporting --> Valuation`.

On the :guilabel:`Stock Valuation` report, click the search bar, and in the :guilabel:`Group By`
section of the resulting drop-down menu, select :guilabel:`Lot/Serial number`.

.. tip::
   Click the :icon:`fa-plus` :guilabel:`(+)` icon to the right of a collapsed lot number line to
   :ref:`manually modify the cost <inventory/warehouses_storage/update-unit-price>`.

   This is useful for adjusting individual lot prices when a purchase order or bill includes
   multiple lots/serial numbers, as initial prices are identical upon reception.

.. image:: valuation_by_lots/stock-valuation.png
   :align: center
   :alt: Show valuation report, by lots.

Valuation smart button
----------------------

To access a filtered part of the *Stock Valuation* report specific to a lot or serial number, go to
:menuselection:`Inventory app --> Products --> Lots/Serial Numbers` and select the desired item.

On a lot or serial number page and click the :guilabel:`Valuation` smart button.

> 

Reviewer: I can't see this button on anything in Runbot, so perhaps you can explain what they will see and show an example of what is shown when you click the button?

Author: Added an image! :D

   So, when the quantity is updated from `1.00` to `2.00`, the additional quantity is also valued at
   `$10`, reflecting the latest valuation layer for `LOT 1`.

   .. figure:: valuation_by_lots/existing.png
      :align: center
      :alt: Show valuation of LOT 1 getting updated.

      The inventory adjustment (top line) is valued the same as LOT 1 (bottom line).

.. _inventory/product_management/view-valuation:

View valuation
==============

To find the average cost of a specific lot/serial number, go to :menuselection:`Inventory app -->
Products --> Lots/Serial Numbers` and select the desired record.

Both the :guilabel:`Cost` and :guilabel:`Average Cost` fields show a unit's average cost. The
:guilabel:`Total Value` reflects the total on-hand value for that lot/serial number.

.. figure:: valuation_by_lots/lot.png
   :align: center
   :alt: Show cost of the lot/serial number.

   Lot form, displaying **Cost** field. The **Valuation** smart button is in the top-right.

Valuation layers of a lot/serial number can be viewed through the :ref:`valuation report
<inventory/product_management/valuation-report>` or by clicking the lot/serial number's

> :guilabel:`Valuation` smart button. These detailed, line-by-line records can help determine how each

Reviewer 2: I am not seeing any 'Valuation' smart button on the lot/serial number form in the Master runbot I am using for this review. This should be double-checked during your revisions.

![Screenshot 2024-09-30 105936](https://github.com/user-attachments/assets/3f01d956-96d6-4b3c-9396-344e02c9560e)

Author: Thanks for sharing the URL in your screenshot, so I can diagnose the issue (:

In the product category, please make sure the costing method is set to FIFO or AVCO.
![image](https://github.com/user-attachments/assets/d57e620c-fdd4-4a32-b010-e10f545c2723)

![image](https://github.com/user-attachments/assets/26505c72-dd8d-43bd-8d10-a2303ef69d1e)

While I do have the instructions in the doc to make sure to set the costing method correctly, please let me know if you have suggestions on how to make it clearer (:

inventory move of the specific lot/serial number affects its valuation.

.. _inventory/product_management/valuation-report:

Valuation report
----------------

Display the valuation of lots and serial numbers in the database by going to
:menuselection:`Inventory app --> Reporting --> Valuation`.

On the resulting :guilabel:`Stock Valuation` report, click the search bar, and in the
:icon:`oi-group` :guilabel:`Group By` section of the resulting drop-down menu, select
:guilabel:`Lot/Serial number`.

.. tip::
   Click the :icon:`fa-plus` :guilabel:`(plus)` icon to the right of a collapsed lot number line to
   :ref:`manually modify the cost <inventory/warehouses_storage/update-unit-price>`.

   This is useful for adjusting individual lot prices when a purchase order or bill includes

>    multiple lots/serial numbers, as initial prices are identical upon reception.

Reviewer 2: I am not seeing any 'plus' icon to the right of a collapsed lot number line -- at least, not in the Master runbot I am using for this review. Please double-check this during your revisions to ensure the accuracy of this statement.

Author: You'll have to make sure the costing method is set to avco or fifo, and process a receipt, to see the plus button (:

.. image:: valuation_by_lots/stock-valuation.png
   :align: center
   :alt: Show valuation report, by lots.

Valuation smart button
----------------------

> To access a filtered part of the *Stock Valuation* report specific to a lot or serial number, go to

Reviewer 2: ```suggestion
To access a filtered part of the *Stock Valuation* report, specific to a lot or serial number, go to
```

> :menuselection:`Inventory app --> Products --> Lots/Serial Numbers` and select the desired item.

Reviewer 2: ```suggestion
:menuselection:`Inventory app --> Products --> Lots/Serial Numbers`, and select the desired item.
```

> On the :guilabel:`Lot/Serial Numbers` page, click the :guilabel:`Valuation` smart button.

Reviewer 2: Again -- this smart button is **not** appearing in the Master runbot I am using. Since LARM encountered the same issue, you will need to double-check this, and determine why it hasn't been showing up for either of us. Once that's figured out. please add any necessary elaboration/explanation as to why that may/may not show up.

Author: I'm quite sure the reason the smart button is not showing up is because FIFO or AVCO wasn't selected as a costing method. Let me know if this issue persists in your second run-through of the doc, and I'd be happy to jump in

:menuselection:`Inventory app --> Reporting --> Valuation`.

On the resulting :guilabel:`Stock Valuation` report, click the search bar, and in the
:icon:`oi-group` :guilabel:`Group By` section of the resulting drop-down menu, select
:guilabel:`Lot/Serial number`.

.. tip::
   Click the :icon:`fa-plus` :guilabel:`(plus)` icon to the right of a collapsed lot number line to
   :ref:`manually modify the cost <inventory/warehouses_storage/update-unit-price>`.

   This is useful for adjusting individual lot prices when a purchase order or bill includes
   multiple lots/serial numbers, as initial prices are identical upon reception.

.. image:: valuation_by_lots/stock-valuation.png
   :align: center
   :alt: Show valuation report, by lots.

Valuation smart button
----------------------

To access a filtered part of the *Stock Valuation* report, specific to a lot or serial number, go to
:menuselection:`Inventory app --> Products --> Lots/Serial Numbers`, and select the desired item.

On the :guilabel:`Lot/Serial Numbers` page, click the :guilabel:`Valuation` smart button.

.. figure:: valuation_by_lots/lot-stock-valuation.png

>    :align: center

Reviewer 3: ```suggestion
```
